packages/concerto-core/lib/introspect/modelfile.js

```diff
@@ -583,19 +583,28 @@ class ModelFile extends Decorated {
         return this.getDeclarations(EnumDeclaration);
     }
 
+    /**
+     * Get the ScalarDeclaration defined in this ModelFile
+     * @return {ScalarDeclaration[]} the ScalarDeclaration defined in the model file
+     */
+    getScalarDeclarations() {
+        return this.getDeclarations(ScalarDeclaration);
+    }
+
     /**
      * Get the instances of a given type in this ModelFile
      * @param {Function} type - the type of the declaration
-     * @return {ClassDeclaration[]} the ClassDeclaration defined in the model file
+     * @return {Object[]} the ClassDeclaration defined in the model file
      */
     getDeclarations(type) {
         let result = [];
         for(let n=0; n < this.declarations.length; n++) {
-            let classDeclaration = this.declarations[n];
-            if(classDeclaration instanceof type) {
-                result.push(classDeclaration);
+            let declaration = this.declarations[n];
+            if(declaration instanceof type && !Object.is(declaration, {})) {
```

Comment:
I don't think this `Object.is` condition will ever be anything other than false, because it only returns true if two values point to the same object in memory. Why would an empty object declaration exist in this list anyway?

From the MDN docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is

```
const foo = { a: 1 };
const bar = { a: 1 };
const sameFoo = foo;
Object.is(foo, foo); // true
Object.is(foo, bar); // false
```